lua/lvim/core/bufferline.lua

```diff
@@ -1,18 +1,127 @@
+      close_command = "bdelete! %d", -- can be a string | function, see "Mouse actions"
+      right_mouse_command = "vert sbuffer %d", -- can be a string | function, see "Mouse actions"
+      left_mouse_command = "buffer %d", -- can be a string | function, see "Mouse actions"
+      middle_mouse_command = nil, -- can be a string | function, see "Mouse actions"
+      -- NOTE: this plugin is designed with this icon in mind,
+      -- and so changing this is NOT recommended, this is intended
+      -- as an escape hatch for people who cannot bear it for whatever reason
+      indicator_icon = "▎",
+      buffer_close_icon = "",
+      modified_icon = "●",
+      close_icon = "",
+      left_trunc_marker = "",
+      right_trunc_marker = "",
+      --- name_formatter can be used to change the buffer's label in the bufferline.
+      --- Please note some names can/will break the
+      --- bufferline so use this at your discretion knowing that it has
+      --- some limitations that will *NOT* be fixed.
+      name_formatter = function(buf) -- buf contains a "name", "path" and "bufnr"
+        -- remove extension from markdown files for example
+        if buf.name:match "%.md" then
+          return vim.fn.fnamemodify(buf.name, ":t:r")
+        end
+      end,
+      max_name_length = 18,
+      max_prefix_length = 15, -- prefix used when a buffer is de-duplicated
+      tab_size = 18,
+      diagnostics = "nvim_lsp",
+      diagnostics_update_in_insert = false,
+      diagnostics_indicator = diagnostics_indicator,
+      -- NOTE: this will be called a lot so don't do any heavy processing here
+      custom_filter = custom_filter,
+      offsets = {
+        {
+          filetype = "undotree",
+          text = "Undotree",
+          highlight = "PanelHeading",
+          padding = 1,
+        },
+        {
+          filetype = "NvimTree",
+          text = "Explorer",
+          highlight = "PanelHeading",
+          padding = 1,
+        },
+        {
+          filetype = "DiffviewFiles",
+          text = "Diff View",
+          highlight = "PanelHeading",
+          padding = 1,
+        },
+        {
+          filetype = "flutterToolsOutline",
+          text = "Flutter Outline",
+          highlight = "PanelHeading",
+        },
+        {
+          filetype = "packer",
+          text = "Packer",
+          highlight = "PanelHeading",
+          padding = 1,
+        },
+      },
+      show_buffer_icons = true, -- disable filetype icons for buffers
+      show_buffer_close_icons = true,
+      show_close_icon = false,
+      show_tab_indicators = true,
+      persist_buffer_sort = true, -- whether or not custom sorted buffers should persist
+      -- can also be a table containing 2 custom separators
+      -- [focused and unfocused]. eg: { '|', '|' }
+      separator_style = "thin",
+      enforce_regular_tabs = false,
+      always_show_bufferline = false,
```

Comment:
This one might be a bit confusing, I even forgot about it. I think it controls whether a single buffer should still show the bufferline. If it's set to `false`, then opening a buffer from Alpha won't show the bufferline.

Reply:
yes which is what i intended to do and what i think was missing from barbar

Reply:
Sounds good, and people can always change it.

--- 

NB: Please edit the description to include https://github.com/LunarVim/LunarVim/issues/2100 as well :)